dist/stat_data.py

```diff
@@ -446,19 +446,41 @@ def __init__(self, name, desc, flags=''):
     TxnStat('txn_commit_queue_inserts', 'transactions commit timestamp queue inserts total'),
     TxnStat('txn_commit_queue_len', 'transactions commit timestamp queue length'),
+    TxnStat('txn_commit_queue_rdlock', 'transactions commit timestamp queue read locks'),
+    TxnStat('txn_commit_queue_try_fail', 'transactions commit timestamp queue write locks failed'),
+    TxnStat('txn_commit_queue_try_success', 'transactions commit timestamp queue write locks succeeded'),
+    TxnStat('txn_commit_queue_wrlock', 'transactions commit timestamp queue write locks'),
     TxnStat('txn_fail_cache', 'transaction failures due to cache overflow'),
+    TxnStat('txn_global_rdlock', 'transactions global read locks'),
+    TxnStat('txn_global_wrlock', 'transactions global write locks'),
+    TxnStat('txn_global_try_fail', 'transactions global write locks failed'),
+    TxnStat('txn_global_try_success', 'transactions global write locks succeeded'),
     TxnStat('txn_pinned_checkpoint_range', 'transaction range of IDs currently pinned by a checkpoint', 'no_clear,no_scale'),
     TxnStat('txn_pinned_range', 'transaction range of IDs currently pinned', 'no_clear,no_scale'),
     TxnStat('txn_pinned_snapshot_range', 'transaction range of IDs currently pinned by named snapshots', 'no_clear,no_scale'),
     TxnStat('txn_pinned_timestamp', 'transaction range of timestamps currently pinned', 'no_clear,no_scale'),
     TxnStat('txn_pinned_timestamp_oldest', 'transaction range of timestamps pinned by the oldest timestamp', 'no_clear,no_scale'),
+    TxnStat('txn_query_ts', 'transactions query timestamp calls'),
+    TxnStat('txn_read_queue_empty', 'transactions read timestamp queue insert to empty'),
     TxnStat('txn_read_queue_head', 'transactions read timestamp queue inserts to head'),
     TxnStat('txn_read_queue_inserts', 'transactions read timestamp queue inserts total'),
     TxnStat('txn_read_queue_len', 'transactions read timestamp queue length'),
+    TxnStat('txn_read_queue_rdlock', 'transactions read timestamp queue read locks'),
+    TxnStat('txn_read_queue_try_fail', 'transactions read timestamp queue write locks failed'),
+    TxnStat('txn_read_queue_try_success', 'transactions read timestamp queue write locks succeeded'),
+    TxnStat('txn_read_queue_wrlock', 'transactions read timestamp queue write locks'),
     TxnStat('txn_rollback', 'transactions rolled back'),
+    TxnStat('txn_set_ts', 'transactions set timestamp calls'),
+    TxnStat('txn_setts_commit', 'transactions set timestamp commit calls'),
+    TxnStat('txn_setts_commit_upd', 'transactions set timestamp commit updates'),
+    TxnStat('txn_setts_oldest', 'transactions set timestamp oldest calls'),
+    TxnStat('txn_setts_oldest_upd', 'transactions set timestamp oldest updates'),
+    TxnStat('txn_setts_stable', 'transactions set timestamp stable calls'),
+    TxnStat('txn_setts_stable_upd', 'transactions set timestamp stable updates'),
```

Comment:
I'd really like these statistics, but I found the names a bit confusing at the start. I'd prefer "transaction global .." rather than "transactions" - when I read transactions it made me think these statistics were tracking live transactions with timestamps associated with them, rather than calls to the set_timestamp API.

Reply:
I'm not a fan of `setts`, I strongly prefer `set_ts`.